File: CSC110-Cpp/_solutions/Module09-Introduction-to-Arrays/debug03/reverse_array.cpp
// reverse_array.cpp
// Fills an array from user input, reverses it in place using a helper
// function, and prints the array before and after.
#include <iostream>
using namespace std;

void printArray(int arr[], int size) {
    for (int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

// Reverses the array in place by swapping elements from the two ends
// toward the middle.
void reverseArray(int arr[], int size) {
    for (int i = 0; i < size / 2; i++) {
        int temp = arr[i];
        arr[i] = arr[size - 1 - i];
        arr[size - 1 - i] = temp;
    }
}

int main() {
    const int SIZE = 6;
    int values[SIZE];

    cout << "Enter " << SIZE << " integers: ";
    for (int i = 0; i < SIZE; i++) {
        cin >> values[i];
    }

    cout << "Original: ";
    printArray(values, SIZE);

    reverseArray(values, SIZE);

    cout << "Reversed: ";
    printArray(values, SIZE);
    return 0;
}

File: CSC110-Java/_solutions/Module09-Introduction-to-Arrays/debug03/ReverseArray.java
// ReverseArray.java
// Reverses an array of integers in place (without creating a second
// array) by swapping elements from the two ends toward the middle.

public class ReverseArray {
    public static void main(String[] args) {
        int[] values = {3, 8, 12, 5, 21, 7};

        System.out.print("Original: ");
        printArray(values);

        reverse(values);

        System.out.print("Reversed: ");
        printArray(values);
    }

    // Swaps values[i] with its mirror element from the other end.
    public static void reverse(int[] arr) {
        for (int i = 0; i < arr.length / 2; i++) {
            int mirror = arr.length - 1 - i;
            int temp = arr[i];
            arr[i] = arr[mirror];
            arr[mirror] = temp;
        }
    }

    public static void printArray(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i]);
            if (i < arr.length - 1) {
                System.out.print(", ");
            }
        }
        System.out.println();
    }
}
